### PR TITLE
[Merged by Bors] - feat: Binary products of finite measures and probability measures.

### PR DESCRIPTION
**Pull request by kkytola**
[Upstreaming from PFR project](https://teorth.github.io/pfr/docs/PFR/ForMathlib/FiniteMeasureProd.html)

This PR defines binary products of finite measures and probability measures. 

---

 * In PFR project, the continuous dependence of a product measure on the factors (in the topology of weak convergence) was shown and applied for probability measures on finite types. This continuity is not included in the PR: in Mathlib the plan is to later show that product measures depend continuously on their factors when the underlying space is metrizable and separable.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+lemma mass_prod : (μ.prod ν).mass = μ.mass * ν.mass := by
+  simp only [mass, univ_prod_univ.symm, toMeasure_prod]
+  rw [← ENNReal.toNNReal_mul]
+  exact congr_arg ENNReal.toNNReal (Measure.prod_prod univ univ)
+
+lemma zero_prod : (0 : FiniteMeasure α).prod ν = 0 := by
+  rw [← mass_zero_iff, mass_prod, zero_mass, zero_mul]
+
+lemma prod_zero : μ.prod (0 : FiniteMeasure β) = 0 := by
+  rw [← mass_zero_iff, mass_prod, zero_mass, mul_zero]
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:68]: shouldn't these lemmas be simp lemmas?

File: Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+section ProbabilityMeasure_product
+
+namespace ProbabilityMeasure
+
+variable {α : Type*} [MeasurableSpace α] {β : Type*} [MeasurableSpace β]
+
+/-- The binary product of probability measures. -/
+noncomputable def prod (μ : ProbabilityMeasure α) (ν : ProbabilityMeasure β) :
+    ProbabilityMeasure (α × β) :=
+  ⟨μ.toMeasure.prod ν.toMeasure, Measure.prod.instIsProbabilityMeasure μ.toMeasure ν.toMeasure⟩
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/FiniteMeasureProd.lean:111]:
```suggestion
  ⟨μ.toMeasure.prod ν.toMeasure, by infer_instance⟩
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by kkytola** [comment]
> ✌️ kkytola can now approve this pull request. To approve and merge a pull request, simply reply with `bors r+`. More detailed instructions are available [here](https://bors.tech/documentation/getting-started/#reviewing-pull-requests).

bors r+

**Comment by kkytola** [comment]
> ✌️ kkytola can now approve this pull request. To approve and merge a pull request, simply reply with `bors r+`. More detailed instructions are available [here](https://bors.tech/documentation/getting-started/#reviewing-pull-requests).

bors r+

**Pull request closed without merging**